pkg/apis/alibabacloudprovider/v1beta1/alibabacloudmachineproviderconfig_types.go

```diff
@@ -58,186 +62,114 @@ type AlibabaCloudMachineProviderConfig struct {
-	KeyPairName string `json:"keyPairName,omitempty"`
+	//Tags are metadata information about the instance
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
 
 	//The name of the instance Resource Access Management (RAM) role. You can call the ListRoles operation provided by RAM to query the instance RAM roles that you have created.
+	//+optional
 	RAMRoleName string `json:"ramRoleName,omitempty"`
 
-	//
 	//Specifies whether to enable security hardening. Valid values:
-	//
 	//Active: enables security hardening. This value is applicable only to public images.
 	//Deactive: does not enable security hardening. This value is applicable to all images.
+	// +optional
 	SecurityEnhancementStrategy string `json:"securityEnhancementStrategy,omitempty"`
 
 	//The ID of the resource group to which to assign the instance.
+	// +optional
 	ResourceGroupID string `json:"resourceGroupId,omitempty"`
 
 	//The release protection property of the instance. It specifies whether you can use the ECS console or call the DeleteInstance operation to manually release the instance. Default value: false. Valid values:
-	//
 	//true: enables release protection.
 	//false: disables release protection.
+	// +optional
 	DeletionProtection bool `json:"deletionProtection,omitempty"`
 
 	//Specifies whether to associate the instance on a dedicated host with the dedicated host. Valid values:
-	//
 	//default: does not associate the instance with the dedicated host. When you restart an instance in the No Fees for Stopped Instances (VPC-Connected) state, the instance is automatically deployed to another dedicated host in the automatic deployment resource pool if the available resources of the original dedicated host are insufficient.
 	//host: associates the instance with the dedicated host. When you restart an instance in the No Fees for Stopped Instances (VPC-Connected) state, the instance still resides on the original dedicated host. If the available resources of the original dedicated host are insufficient, the instance fails to restart.
 	//Default value: default.
+	// +optional
 	Affinity string `json:"affinity,omitempty"`
 
 	//Specifies whether to create the instance on a dedicated host. Valid values:
-	//
 	//default: creates the instance on a non-dedicated host.
 	//host: creates the instance on a dedicated host. If you do not specify the DedicatedHostID parameter, Alibaba Cloud automatically selects a dedicated host for the instance.
 	//Default value: default.
+	// +optional
 	Tenancy InstanceTenancy `json:"tenancy,omitempty"`
 
 	//The ID of the dedicated host on which to create the instance.
-	//
 	//You can call the DescribeDedicatedHosts operation to query the dedicated host list.
-	//
 	//When the DedicatedHostID parameter is specified, the SpotStrategy and SpotPriceLimit parameters are ignored. This is because preemptible instances cannot be created on dedicated hosts.
+	// +optional
 	DedicatedHostID string `json:"dedicatedHostId"`
```

Comment:
We should check how we do this on AWS, wondering if we group this into a struct there 🤔

Reply:
I didn't see a dedicated host ID. Only information regarding tenancy.